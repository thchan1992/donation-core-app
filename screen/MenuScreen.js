import { useState, useEffect } from "react";
import { View, Text, StyleSheet, ScrollView } from "react-native";
import { selectedCharity } from "../data/charityData";
import { FlatList } from "react-native-gesture-handler";
import PrimaryButton from "../components/PrimaryButton";
import { beneficiaryList, charityList } from "../data/charityData";
import BeneficiaryListItem from "../components/BeneficiaryListItem";

const MenuScreen = ({ navigation }) => {
  const renderBeneficiaryList = (itemData) => {
    return (
      <BeneficiaryListItem
        name={itemData.item.beneficiary}
        imageUrl={itemData.item.imageUrl}
        onPress={() => {
          const arr = charityList.filter((char) => {
            return char.beneficiary == itemData.item.beneficiary;
          });
          navigation.navigate("CharityListScreen", { charityList: arr });
        }}
      />
    );
  };

  return (
    <View style={styles.container}>
      <Text style={styles.instrucText}>
        Choose who you would like to donate
      </Text>
      <FlatList
        data={beneficiaryList}
        keyExtractor={(bene) => bene.beneficiary}
        renderItem={renderBeneficiaryList}
        numColumns={1}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  instrucText: {
    alignSelf: "center",
    padding: 5,
    fontWeight: "bold",
  },
});

export default MenuScreen;
