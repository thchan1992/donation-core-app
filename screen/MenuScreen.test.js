import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";
import MenuScreen from "./MenuScreen";

describe("MenuScreen", () => {
  it("should match the snapshot", () => {
    const component = renderer.create(<MenuScreen />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
