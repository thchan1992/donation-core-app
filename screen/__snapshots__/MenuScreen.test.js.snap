// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuScreen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "padding": 5,
      }
    }
  >
    Please choose who you like to help?
  </Text>
  <RCTScrollView
    collapsable={false}
    data={
      Array [
        Object {
          "beneficiary": "disabled",
          "imageUrl": 1,
        },
        Object {
          "beneficiary": "wild animal",
          "imageUrl": 1,
        },
        Object {
          "beneficiary": "cancer patient",
          "imageUrl": 1,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#28C7FA",
              "borderColor": "#FF304F",
              "borderRadius": 8,
              "borderWidth": 1.5,
              "height": 180,
              "margin": 5,
              "opacity": 1,
              "width": 200,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#FF304F",
                "borderRadius": 8,
                "borderWidth": 1.5,
                "height": 150,
                "justifyContent": "center",
                "width": 200,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 2.5,
                "textTransform": "uppercase",
              }
            }
          >
            disabled
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#28C7FA",
              "borderColor": "#FF304F",
              "borderRadius": 8,
              "borderWidth": 1.5,
              "height": 180,
              "margin": 5,
              "opacity": 1,
              "width": 200,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#FF304F",
                "borderRadius": 8,
                "borderWidth": 1.5,
                "height": 150,
                "justifyContent": "center",
                "width": 200,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 2.5,
                "textTransform": "uppercase",
              }
            }
          >
            wild animal
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#28C7FA",
              "borderColor": "#FF304F",
              "borderRadius": 8,
              "borderWidth": 1.5,
              "height": 180,
              "margin": 5,
              "opacity": 1,
              "width": 200,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#FF304F",
                "borderRadius": 8,
                "borderWidth": 1.5,
                "height": 150,
                "justifyContent": "center",
                "width": 200,
              }
            }
          />
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 2.5,
                "textTransform": "uppercase",
              }
            }
          >
            cancer patient
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
